samza-core/src/test/scala/org/apache/samza/storage/TestTransactionalStateTaskStorageManager.java

```diff
@@ -492,6 +493,32 @@ public void testWriteChangelogOffsetFilesWithNoChangelogOffset() throws IOExcept
     fail("Should have thrown an exception if no changelog offset found for checkpointed store");
   }
 
+  @Test
+  public void testRemoveOldCheckpoints() {
```

Comment:
Looks like this is testing the specific sub-case when listFiles returns null. If so, let's rename method to reflect that (e.g. testRemoveOldCheckpointsWhenXYZ

Reply:
done